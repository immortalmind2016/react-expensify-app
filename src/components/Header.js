import React from "react"
import {NavLink} from "react-router-dom"
import {connect} from "react-redux"
import {startLogout} from "../actions/auth"
export const  Header=({startLogout})=>(
    <header>
        <h1>Expensify</h1>
        <div>
            <NavLink exact={true} to="/dashboard" activeClassName="is-active"> dashboard </NavLink>
            <NavLink to="/create" activeClassName="is-active"> Create </NavLink>
            <NavLink to="/edit/55" activeClassName="is-active"> edit </NavLink>
            <NavLink to="/help" activeClassName="is-active"> help </NavLink>
            <button onClick={startLogout}>Logout</button>
        </div>
    </header>
)

const mapDispatchToProps=(dispatch)=>({
    startLogout:()=>dispatch(startLogout())
})

export default connect(undefined,mapDispatchToProps)(Header)