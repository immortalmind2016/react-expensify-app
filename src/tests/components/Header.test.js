//react-test-render
//import ReactShallowRenderer from "react-test-renderer/shallow" we used Enzyme instead
import React from "react";
import {shallow} from "enzyme"
import Header from "../../components/Header"
import toJson from 'enzyme-to-json';

test("should test Header component",()=>{

const wrapper=shallow(<Header />);
//expect(wrapper.find("h1").text()).toBe("Expensify")

//expect(toJson(wrapper)).toMatchSnapshot();  
// now i can use it without toJson instead because i'm configurated my jest.config.json file
expect(wrapper).toMatchSnapshot();  

})


//test("should render header correctly ",()=>{
  //  const renderer=new ReactShallowRenderer();
   // renderer.render(<Header />)
   // expect(renderer.getRenderOutput()).toMatchSnapshot() 
    // will generate snapshot to my Component and save it
    // next time i can accept new changes or now
    // we will use enzyme because it's difcult to use renderer with complex components
    // raf Libirary -> request animation frame ( for browser )

//})