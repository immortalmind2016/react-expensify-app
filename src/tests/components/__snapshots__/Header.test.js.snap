// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test Header component 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <div>
    <NavLink
      activeClassName="is-active"
      aria-current="page"
      exact={true}
      to="/dashboard"
    >
       dashboard 
    </NavLink>
    <NavLink
      activeClassName="is-active"
      aria-current="page"
      to="/create"
    >
       Create 
    </NavLink>
    <NavLink
      activeClassName="is-active"
      aria-current="page"
      to="/edit/55"
    >
       edit 
    </NavLink>
    <NavLink
      activeClassName="is-active"
      aria-current="page"
      to="/help"
    >
       help 
    </NavLink>
    <button
      onClick={[Function]}
    >
      Logout
    </button>
  </div>
</header>
`;
